bootstrap/src/main/java/jenkins/bootstrap/BootLogic.java

```diff
@@ -0,0 +1,21 @@
+package jenkins.bootstrap;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.util.ServiceLoader;
+
+/**
+ * Interface for code that boots up Jenkins.
+ *
+ * <p>
+ * Bootstrap code uses {@link ServiceLoader} to look for subtypes of this marker interface
+ * and invokes {@link #contextInitialized(ServletContextEvent)} and {@link #contextDestroyed(ServletContextEvent)}
+ *
+ * @author Kohsuke Kawaguchi
```

Comment:
since Javadocs for new API classes would be useful (since we really want to implement them). Or `@Restricted` otherwise